azavea/cac-tripplanner-android
# Fancy notifications

## Overview

Add 'been' button to geofence notifications; on tap, open detail view and change flag from 'want to go' to 'been.' Add image to notification, where present, with text fallback in case image not available or fails to load for some reason. (It should always exist, though).

Also expands Crashlytics logging. Note that because we are using Crashlytics with Firebase integration, custom events are not supported, so simple log messages are sent for geofence add/remove events instead.

Also updates documentation and files for setting various app service keys for Fabric/Crashlytics support.

## Demo

![image](https://user-images.githubusercontent.com/960264/41604647-bb6d8cbc-73ad-11e8-8c7c-fb4ec5a12172.png)



## Notes

Adds some logic for notification grouping, but currently notifications will only group with the auto-grouping behavior, which happens when there are four or more. To logically group them for fewer notices would require a single notification be sent and then subsequently updated with the same identifier and with a shared title, which might make sense to do for multiple events occurring at the same location. But as that would require significant additional effort to support and will likely be implemented very differently once [support for custom locations on events](azavea/cac-tripplanner#1008) has been added, this PR does not do that. We may want to add grouping events by destination once that has been done.

## Testing

 - Mark a place or event with an associated place as 'want to go'
 - Use emulator location mock to set location to at or near the place
 - Open Maps in emulator to cause location to update
 - Notification should appear within a few minutes with the image as above
 - In-notification 'been' button should function as expected

Closes #46.
